### PR TITLE
Add reason to error responses

### PR DESCRIPTION
**Pull request by LucasCairns**
- Add reason to response when there is an OASys permission issue
- Add reason to response when therre is a duplicate offender record
- Update permission error to use 403 response

```json
{
    "status": 403,
    "developerMessage": "User does not have permisson...",
    "reason": "OASYS_PERMISSION"
}
```

```json
{
    "status": 400,
    "developerMessage": "Duplicate offender...",
    "reason": "DUPLICATE_OFFENDER_RECORD"
}
```

[Relates to the work in PR 153 of the UI](https://github.com/ministryofjustice/hmpps-risk-assessment-ui/pull/153]) and should be merged first

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/assessments/api/ErrorResponseOffender.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.justice.digital.assessments.api
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.swagger.v3.oas.annotations.media.Schema
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+data class ErrorResponseOffender(
+  @Schema(required = true, description = "Fullname of the offender", example = "Ray Arnold")
+  val name: String? = null,
```
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/assessments/api/ErrorResponseOffender.kt:9]: is the full name GDPR compliant? can we include things like this in logs/errors? might worth double checking

Comment (LucasCairns) [reply src/main/kotlin/uk/gov/justice/digital/assessments/api/ErrorResponseOffender.kt:9]: Good point - we require the context to be returned to the frontend in order to display as part of the error message. Will do a pass to ensure this isn't passed in to the logs

File: src/main/kotlin/uk/gov/justice/digital/assessments/restclient/AssessmentUpdateRestClient.kt [hunk 2]
```diff
@@ -57,38 +57,45 @@ class AssessmentUpdateRestClient {
       }
   }
 
+  data class OffenderContext (
```
Comment (LucasCairns) [inline src/main/kotlin/uk/gov/justice/digital/assessments/restclient/AssessmentUpdateRestClient.kt:60]: Noticed we would be passing multiple parts of the offender through here, so created an object to contain this information. Not quite sure where this should live though.. or this was the correct thing to do - open to suggestions 🤔

File: src/main/kotlin/uk/gov/justice/digital/assessments/api/ErrorResponseOffender.kt (hunk 1 above)
Comment (LucasCairns) [reply src/main/kotlin/uk/gov/justice/digital/assessments/api/ErrorResponseOffender.kt]: Removed this code - agreed to split this out as a separate ticket, we may be able to get this information back in the OASys error response

File: src/main/kotlin/uk/gov/justice/digital/assessments/restclient/AssessmentUpdateRestClient.kt (hunk 2 above)
Comment (LucasCairns) [reply src/main/kotlin/uk/gov/justice/digital/assessments/restclient/AssessmentUpdateRestClient.kt]: Removed this code - as per above

**Review by adelaidaa — APPROVED** [review]
Nice one!

**Pull request merged**